templates/client-clusterrole.yaml

```diff
@@ -8,13 +8,22 @@ metadata:
     chart: {{ template "consul.chart" . }}
     heritage: {{ .Release.Service }}
     release: {{ .Release.Name }}
-{{- if .Values.global.enablePodSecurityPolicies }}
+{{- if (or .Values.global.enablePodSecurityPolicies .Values.global.bootstrapACLs) }}
 rules:
+{{- end }}
+{{- if .Values.global.enablePodSecurityPolicies }}
   - apiGroups: ["policy"]
     resources: ["podsecuritypolicies"]
     resourceNames:
     - {{ template "consul.fullname" . }}-client
     verbs:
     - use
 {{- end }}
+{{- if .Values.global.bootstrapACLs }}
+  - apiGroups: [""]
+    resources:
```

Comment:
If possible it might be good to figure out how to scope this to just _your_ secrets using `resourceNames` instead of "all secrets": https://kubernetes.io/docs/reference/access-authn-authz/rbac/#referring-to-resources

Reply:
Restricted the client and catalog sync roles to only refer to their specific secret.